discretize/CylMesh.py

```diff
@@ -123,72 +227,531 @@ def vectorCCx(self):
+
+    @property
+    def _ishangingEy(self):
+        if getattr(self, '__ishangingEy', None) is None:
+            hang_x = np.zeros(self._ntNx, dtype=bool)
+            hang_x[0] = True
+            self.__ishangingEy = np.kron(
+                np.ones(self._ntNz, dtype=bool),
+                np.kron(
+                    np.ones(self.nCy, dtype=bool),
+                    hang_x
+                )
+            )
+        return self.__ishangingEy
+
+    @property
+    def _hangingEy(self):
+        if getattr(self, '__hangingEy', None) is None:
+            self.__hangingEy = dict(zip(
+                np.nonzero(self._ishangingEy)[0].tolist(),
+                [None]*len(self.__ishangingEy))
+            )
+        return self.__hangingEy
+
+    @property
+    def _axis_of_symmetry_Ez(self):
+        if getattr(self, '__axis_of_symmetry_Ez', None) is None:
+            axis_x = np.zeros(self._ntNx, dtype=bool)
+            axis_x[0] = True
+
+            axis_y = np.zeros(self._ntNy, dtype=bool)
+            axis_y[0] = True
+            self.__axis_of_symmetry_Ez = np.kron(
+                np.ones(self.nCz, dtype=bool),
+                np.kron(
+                    axis_y,
+                    axis_x
+                )
+            )
+        return self.__axis_of_symmetry_Ez
+
+    @property
+    def _ishangingEz(self):
+        if getattr(self, '__ishangingEz', None) is None:
+            if self.isSymmetric:
+                self.__ishangingEz = np.ones(self._ntEz, dtype=bool)
+            else:
+                hang_x = np.zeros(self._ntNx, dtype=bool)
+                hang_x[0] = True
+
+                hang_y = np.zeros(self._ntNy, dtype=bool)
+                hang_y[-1] = True
+
+                hangingEz = np.kron(
+                    np.ones(self.nCz, dtype=bool),
+                    (
+                        np.kron(
+                            np.ones(self._ntNy, dtype=bool),
+                            hang_x
+                        ) |
+                        np.kron(
+                            hang_y,
+                            np.ones(self._ntNx, dtype=bool)
+                        )
+                    )
+                )
+
+                self.__ishangingEz = hangingEz & ~self._axis_of_symmetry_Ez
+
+        return self.__ishangingEz
+
+    @property
+    def _hangingEz(self):
+        if getattr(self, '__hangingEz', None) is None:
+            # deflate
+            deflateEz = np.hstack([
+                np.hstack([
+                    np.zeros(self._ntNy-1, dtype=int),
+                    np.arange(1, self._ntNx, dtype=int)
+                ]) +
+                i*int(self._ntNx*self._ntNy)
+                for i in range(self.nCz)
+            ])
+            deflate = zip(
+                np.nonzero(self._ishangingEz)[0].tolist(), deflateEz
+            )
+
+            self.__hangingEz = dict(deflate)
+        return self.__hangingEz
+
+    @property
+    def _axis_of_symmetry_N(self):
```

Comment:
I am not entirely sure what this is for. Can you add a doc string?

Reply:
- [ ] todo: docstring here. This identifies the nodes along the axis of symmetry that need to be eliminated